bootstrap_cfn/iam.py

```diff
@@ -18,18 +20,299 @@ def __init__(self, aws_profile_name, aws_region_name='eu-west-1'):
+                exists = True
+                logging.info("IAM::check_remote_certificate_exists: "
+                             "Local and remote certificates are equal, "
+                             "certificate id '%s' "
+                             % (cert_name))
+            else:
+                logging.info("IAM::check_remote_certificate_exists: "
+                             "Local and remote certificates are not the same, "
+                             "certificate id '%s' "
+                             % (cert_id))
+                exists = False
+        # Handle any problems connecting to the remote AWS 
+        except AWSQueryConnection.ResponseError as error:
+                    exists = False
+                    logging.info("IAM::check_remote_certificate_exists: "
+                                 "Could not find certificate '%s': "
+                                 "Error %s - %s" % (cert_id,
+                                                    error.status,
+                                                    error.reason))
+        return exists
+
+    def compare_certificate_data(self, cert_data1, cert_data2):
+        """
+        Compare two sets of certificate data for equality
+
+        Args:
+            cert1(dictionary): Dictionary of certificate data,
+                with certs, chains and keys
+            cert2(dictionary): Dictionary of certificate data,
+                with certs, chains and keys
+
+        Returns:
+            are_equal: True if the certficate data are equal,
+            false otherwise
+        """
+
+        are_equal = False
+
+        start_text = "-----BEGIN CERTIFICATE-----"
+        end_text = "-----END CERTIFICATE-----"
+        certs_are_equal = self.compare_text_between_strings(
+                                                                 cert_data1.get("cert", None),
+                                                                 cert_data2.get("cert", None),
+                                                                 start_text,
+                                                                 end_text)
+        if not certs_are_equal:
+            logging
+        else:
+            start_text = "-----BEGIN CERTIFICATE-----"
+            end_text = "-----END CERTIFICATE-----"
+            chains_are_equal = self.compare_text_between_strings(
+                                                          cert_data1.get("chain", None),
+                                                          cert_data2.get("chain", None),
+                                                          start_text,
+                                                          end_text)
+            if not chains_are_equal:
+                logging.info("IAM::compare_certificate_data: "
+                             "Certificate chain data is not equal")
+            else:
+                are_equal = True
+
+        return are_equal
+
+    def compare_text_between_strings(self,
```

Comment:
Can we put the start text/end text in this method (rather than passing it in). Think you might be trying to keep it generic but in this case I feel like it would be better to make it specific to the problem. Maybe rename it to something like compare_certs_body or something like that.